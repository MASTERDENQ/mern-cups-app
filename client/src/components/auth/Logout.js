import React, { Component } from "react";
import { NavLink } from "reactstrap";

class Logout extends Component {
  handle = () => {
    localStorage.setItem("isAuthenticated", false);
  };

  render() {
    return (
      <NavLink href="/test" onClick={this.handle}>
        Logout
      </NavLink>
    );
  }
}

export default Logout;
