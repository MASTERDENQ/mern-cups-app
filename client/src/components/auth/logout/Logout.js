// import React, { Component } from "react";
// import { NavLink } from "reactstrap";

// class Logout extends Component {
//   render() {
//     return (
//       <div>
//         <NavLink href="/">Logout</NavLink>
//       </div>
//     );
//   }
// }

// export default Logout;

import React from "react";
import { NavLink } from "reactstrap";

function Logout({ label }) {
  return (
    <div data-testid="logout">
      <NavLink href="/">{label}</NavLink>
    </div>
  );
}

export default Logout;
