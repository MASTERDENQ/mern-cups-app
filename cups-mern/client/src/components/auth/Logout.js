import React, { Component, Fragment } from "react";
import { NavLink } from "reactstrap";

class Logout extends Component {
  render() {
    return (
      <Fragment>
        <NavLink onClick={this.props.logout} href="#">
          Logout
        </NavLink>
      </Fragment>
    );
  }
}

export default Logout;
