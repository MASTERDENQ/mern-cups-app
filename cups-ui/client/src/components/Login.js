import React from 'react'
import '../css/LoginStyle.css';
import avatar from '../assets/avatar.jpg';


const Login = () => 
    <div className="MainDiv">
        
        <label id="Title">Login</label>

        <img src={avatar} className="Avatar" alt="Avatar" />

        <form className="LoginForm">
            
            <input id="EmailInput" type="email" placeholder="Email Address"></input>

            <input id="PasswordInput" type="password" placeholder="Password"></input>

            <input id="SignatureImageInput" type="file" placeholder="SignatureImage"></input>

            <input id="AudioInput" type="file" placeholder="Audio Input"></input>

            <br/>
            
            <button>Login</button>

        </form>
    </div>

export default Login