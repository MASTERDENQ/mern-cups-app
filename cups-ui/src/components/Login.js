import React from 'react'
import logo from '../assets/avatar.jpg';

const Login = () => <div>
    <img src={logo} className="Avatar" alt="Avatar" />
    <br/><br/>

    <label id="Title">Login</label>

    <form className="LoginForm">
    <input id="EmailInput" type="email" placeholder="Email Address"></input>
    <input id="PasswordInput" type="password" placeholder="Password"></input>
    <input id="SignatureImageInput" type="file" placeholder="SignatureImage"></input>
    <input id="AudioInput" placeholder="Audio Input"></input>
    
    <button>Login</button>
    <button>Create an Account</button>
    </form>
</div>

export default Login