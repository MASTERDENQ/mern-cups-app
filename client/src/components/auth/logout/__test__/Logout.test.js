import React from "react";
import ReactDOM from "react-dom";
import Login from "./../Logout";
import { isTSAanyKeyword } from "@babel/types";
import Logout from "./../Logout";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Logout />, div);
});

it("renders logout Nav Link correctly", () => {
  const { getByTestId } = render(<Logout label="Logout" />);
  expect(getByTestId("logout")).toHaveTextContent("Logout");
});

it("renders logout Nav Link correctly", () => {
  const { getByTestId } = render(<Logout label="save" />);
  expect(getByTestId("logout")).toHaveTextContent("save");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Logout label="save" />).toJSON();
  expect(tree).toMatchSnapshot();
});
