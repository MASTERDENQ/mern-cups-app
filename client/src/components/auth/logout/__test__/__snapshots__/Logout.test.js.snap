// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="logout"
>
  <a
    className="nav-link"
    href="/"
    onClick={[Function]}
  >
    save
  </a>
</div>
`;
